nautobot/dcim/tests/test_forms.py

```diff
@@ -29,23 +28,28 @@ def setUp(self):
     @classmethod
     def setUpTestData(cls):
 
-        site = Site.objects.create(name="Site 1", slug="site-1")
-        rack = Rack.objects.create(name="Rack 1", site=site)
-        manufacturer = Manufacturer.objects.create(name="Manufacturer 1", slug="manufacturer-1")
-        device_type = DeviceType.objects.create(
-            manufacturer=manufacturer,
-            model="Device Type 1",
-            slug="device-type-1",
+        cls.site = Site.objects.create(name="Site 1", slug="site-1")
+        cls.rack = Rack.objects.create(name="Rack 1", site=cls.site)
+
+        # Platforms that have a manufacturer.
+        mfr_platforms = Platform.objects.filter(manufacturer__isnull=False)
+
+        # Get a DeviceType that has a manufacturer in line with one of the platforms.
+        cls.device_type = DeviceType.objects.filter(
+            manufacturer__in=mfr_platforms.values("manufacturer"),
             u_height=1,
-        )
-        device_role = DeviceRole.objects.create(name="Device Role 1", slug="device-role-1", color="ff0000")
-        Platform.objects.create(name="Platform 1", slug="platform-1")
+            is_full_depth=True,
+        ).first()
+        cls.platform = Platform.objects.filter(manufacturer=cls.device_type.manufacturer).first()
+        cls.manufacturer = cls.device_type.manufacturer
```

Comment:
Just being pedantic
```suggestion
        cls.manufacturer = cls.device_type.manufacturer
        cls.platform = Platform.objects.filter(manufacturer=cls.manufacturer).first()
```

Reply:
I thought there was a reason I did this, but I may have fudged it. The logic was that I need to make sure I get a Platform that is in use by the DeviceType for the device, otherwise spurious test failures.